### PR TITLE
Rename `Restart run` mda gui labels to `Select prior ensemble`

### LINKED ISSUE #11920: Gui ESMDA unclear button 'restart run'
### What happened? (You can include a screenshot if it helps explain)

What does it mean to restart a run from a specific ensemble? Will we rerun the selected ensemble?
I think we agreed that it will update from the selected ensemble, and continue running. Maybe `Select prior` would be a better button-text? 
 

Relates to https://github.com/equinor/ert/issues/11856#issuecomment-3328013183n

### PR DESCRIPTION
**Pull request by tbwiik**
**Issue**
Resolves #11920

**Approach**
See #14175, #14190

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'just rapid-tests'`)

## When applicable
- [ ] **When screenshots are changed**: Review screenshot-PR in ert-testdata,
      merge screenshot-PR in ert-testdata **before** merging this PR.
- [x] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Add backport label to latest release (format: 'backport release-branch-name')

### CODE REVIEW

File: src/ert/run_models/multiple_data_assimilation.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def model_post_init(self, ctx: Any) -> None:
         self._parsed_weights = self.parse_weights(self.analysis_settings.weights)
         start_iteration = 0
         total_iterations = len(self._parsed_weights) + 1
-        if self.restart_run:
+        if self.select_prior:
             if not self.prior_ensemble_id:
                 raise ValueError("For restart run, prior ensemble must be set")
```
Comment (SAKavli) [inline src/ert/run_models/multiple_data_assimilation.py]: We discussed in person to raise an issue for reviewing whether this is dead code - as it is untested and seems like an invalid state of the program at first glance.

File: src/ert/run_models/multiple_data_assimilation.py
Comment (SAKavli) [inline src/ert/run_models/multiple_data_assimilation.py:58]: Same with this.

**Review by SAKavli — APPROVED** [review]
Beautiful 🥹

**Comment by tbwiik** [comment]
Cannot add screenshot of changes in any easy way due to access restrictions for personal computer

**Comment by tbwiik** [comment]
might be valuable to add more tests? need someone experienced to confirm or disconfirm this.
Should also resolve the possibly dead path issue above

File: src/ert/run_models/model_factory.py [hunk 2]
```diff
@@ -458,23 +458,20 @@ def _setup_ensemble_information_filter(
...
         # When running from CLI
-        restart_run = args.restart_ensemble_id is not None
-        prior_ensemble = args.restart_ensemble_id or ""
+        prior_ensemble = args.restart_ensemble_id or None
     else:
         # When running from GUI
-        restart_run = args.restart_run
         prior_ensemble = args.prior_ensemble_id
-    return restart_run, prior_ensemble
+    return prior_ensemble
```
Comment (xjules) [inline src/ert/run_models/model_factory.py:472]: nice!

File: src/ert/run_models/multiple_data_assimilation.py [hunk 3]
```diff
@@ -93,7 +91,8 @@ def run_experiment(
             raise ErtRunError("ESMDA does not support restart")
 
         target_experiment = None
-        if self.restart_run:
+
+        if self.prior_ensemble_id:
             id_ = self.prior_ensemble_id
```
Comment (xjules) [inline src/ert/run_models/multiple_data_assimilation.py]: do we  need this `id_` variable?

File: src/ert/storage/migration/to38.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            continue
+
+        index_data = json.loads(index_file.read_text(encoding="utf-8"))
+        experiment_data = index_data.get("experiment", {})
+
+        experiment_type = experiment_data.get("experiment_type", "")
+        if experiment_type != "Multiple Data Assimilation":
+            continue
+
+        experiment_data.pop("restart_run")
```
Comment (xjules) [inline src/ert/storage/migration/to38.py]: We need to be safe here:
```python
    experiment_data.pop("restart_run", None) 
```

File: tests/ert/unit_tests/run_models/snapshots/test_experiment_serialization/test_that_dumped_esmda_matches_snapshot/heat_equationconfig.ert/config.json [hunk 5]
```diff
@@ -599,6 +599,5 @@
     }
   },
   "experiment_type": "Multiple Data Assimilation",
-  "restart_run": false,
   "prior_ensemble_id": null
```
Comment (xjules) [inline tests/ert/unit_tests/run_models/snapshots/test_experiment_serialization/test_that_dumped_esmda_matches_snapshot/heat_equationconfig.ert/config.json:602]: Can it be both, "" and `null`?

**Review by xjules — CHANGES_REQUESTED** [review]
Nice PR @tbwiik ! 
I had only some minor comments.

File: tests/ert/unit_tests/run_models/snapshots/test_experiment_serialization/test_that_dumped_esmda_matches_snapshot/heat_equationconfig.ert/config.json (hunk 5 above)
Comment (tbwiik) [reply tests/ert/unit_tests/run_models/snapshots/test_experiment_serialization/test_that_dumped_esmda_matches_snapshot/heat_equationconfig.ert/config.json:602]: should only be null. Dropped the use of empty string in code for more clear semantics

File: src/ert/run_models/multiple_data_assimilation.py
Comment (tbwiik) [reply src/ert/run_models/multiple_data_assimilation.py:58]: I am 95% sure that this can never be reached. As far as I have understood without an experiment name the configuration is deemed invalid and play button will be disabled

**Review by xjules — APPROVED** [review]
Good job @tbwiik ! 🚀

**Comment by xjules** [comment]
You would need to rebase first though :)

**Comment by xjules** [comment]
> might be valuable to add more tests? need someone experienced to confirm or disconfirm this. Should also resolve the possibly dead path issue above

Since this feature is a bit of a corner case, I think the current tests are fine.

File: src/ert/storage/migration/to38.py [hunk 6]
```diff
@@ -2,61 +2,37 @@
...
-            parquet_file.unlink()
-            logger.info("Moved %s into blob %s", parquet_file, uri)
+        index_data = json.loads(index_file.read_text(encoding="utf-8"))
+        experiment_data = index_data.get("experiment", {})
+
+        experiment_type = experiment_data.get("experiment_type", "")
+        if experiment_type != "Multiple Data Assimilation":
+            continue
+
+        experiment_data.pop("restart_run", None)
```
Comment (SAKavli) [inline src/ert/storage/migration/to38.py]: I was curious if we want to also replace "previous_ensemble_id" with None if it is empty string in the same migration, what do you think mr @xjules ?

Comment (xjules) [reply src/ert/storage/migration/to38.py]: sounds like a good idea. Let's do that. At least we can avoid an extra storage migration.

**Comment by tbwiik** [comment]
testfailure seems like setup flake, and not an actual error in new code
